app/controllers/albums.js

```diff
@@ -1,8 +1,37 @@
-const albumService = require('../services/albums');
+const albumService = require('../services/albums'),
+  errors = require('../errors');
 
 exports.getAllAlbums = (req, res, next) => {
   albumService
     .getAllAlbums()
     .then(albums => res.json({ albums }))
     .catch(next);
 };
+
+exports.buyAlbum = (req, res, next) => {
+  albumService
+    .getAlbumById(req.params.id)
+    .then(album => {
+      if (!album) throw errors.albumNotExists;
+
+      const filter = {
+        id: req.params.id
+      };
+
+      return albumService
+        .getLocalAlbumsByFilter(filter)
+        .then(albums => {
+          const [localAlbum] = albums;
+
+          if (localAlbum && localAlbum.userId === req.body.userId) throw errors.userAlreadyHasTheAlbum;
+
+          if (localAlbum) throw errors.albumAlreadyHasOwnerUser;
```

Comment:
cant a same album be bought by more than one person??

Reply:
No. 
One user can have many albums.
An album can only be from one user.